core/src/main/java/com/xatkit/core/recognition/RecognitionMonitorInflux.java

```diff
@@ -0,0 +1,600 @@
+
+import static java.util.Objects.isNull;
+
+/**
+ * Provides monitoring capabilities for {@link IntentRecognitionProvider}s.
+ * <p>
+ * This class stores analytics information related to intent recognition, and registers a set of REST endpoints
+ * allowing to query them from external applications.
+ * Also, this class manages the connection to an external database (InfluxDB) and stores the information.
+ * Each stored point in the database includes the following information:
+ * <p>
+ * <li><b>bot_id</b>: The ID of the bot that took care of answering the utterance. </li>
+ * <li><b>is_Matched</b>: If said utterance was matched or not. </li>
+ * <li><b>session_id</b>: The ID of the session. </li>
+ * <li><b>origin</b>: The origin of the utterance, assuming it's defined (and the bot runs in different sites). </li>
+ * <li><b>platform</b>: Platform in which the session did ocurr. </li>
+ * <li><b>username</b>: self-explanatory. </li>
+ * <li><b>confidence</b>: How confident is the bot about the answer given to the user. </li>
+ * <li><b>utterance</b>: The received message by the input, which is being interpreted by the bot. </li>
+ * <li><b>intent</b>: Recognized intent by the RecognitionProvider. </li>
+ * <li><b>parameters</b>: Matched parameters </li>
+ * <li><b>timestamp</b>: Time entry when the utterance occurred. </li>
+ * </p>
+ * <p>
+ * The following endpoints can be used to access the stored information:
+ * <ul>
+ * <li><b>/analytics/monitoring</b>: returns a JSON array containing all the persisted monitoring information (note
+ * that this method doesn't support pagination yet, so the returned JSON may be big for long-running applications).</li>
+ * <li><b>/analytics/monitoring/session?sessionId=id</b>: returns a JSON object containing the monitoring information
+ * for the provided {@code sessionId}</li>
+ * <li><b>/analytics/monitoring/unmatched</b>: returns a JSON array containing all the monitoring entries
+ * corresponding to unmatched inputs (i.e. inputs that haven't been successfully translated into intents)</li>
+ * <li><b>/analytics/monitoring/matched</b>: returns a JSON array containing all the monitoring entries
+ * corresponding to matched inputs (i.e. inputs that have been successfully translated into intents)</li>
+ * <li><b>/analytics/monitoring/sessions/stats</b>: returns a JSON object containing computed statistics over
+ * stored sessions (e.g. average time/session, average number of matched inputs/sessions, etc)</li>
+ * <li><b>/analytics/monitoring/origin</b>: returns a JSON object with a resume of the different origins where the data
+ * is received from.</li>
+ * </ul>
+ */
+public class RecognitionMonitorInflux implements RecognitionMonitor {
+
+    /**
+    * The {@link Configuration} key to specify the auth token for the bot to be able to store/query data from an influx bucket.
+    * <p>
+    * This property is mandatory.
+    */
+    static final String INFLUX_TOKEN_KEY = "xatkit.influx.token";
```

Comment:
Most of these constants should be `public static final`: we want to be able to access them from client applications. 
I have also updated some of the constants in RecognitionMonitorMapDB that needed to be public as well.